admin_app/src/app/content/edit/page.tsx

```diff
@@ -20,30 +20,44 @@ export interface Content extends EditContentBody {
 interface EditContentBody {
   content_title: string;
   content_text: string;
-  content_language: string;
+  language_id: number;
   content_metadata: Record<string, unknown>;
 }
 
 const AddEditContentPage = () => {
   const searchParams = useSearchParams();
   const content_id = Number(searchParams.get("content_id")) || null;
-
+  const language_id = Number(searchParams.get("language_id")) || null;
+  const [contentData, setContentData] = React.useState<{ [key: number]: Content }>({});
```

Comment:
To minimize the number of calls to the API, I get the content data as a dictionary with the `language_id` as key and the content as values. This way, when clicking on a different language version we get the corresponding content instead of making a new api call. Reason why we have contentData which is the content dictionary and [content](url) representing the current language version.